### PR TITLE
Macho version

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: src/action.rs [hunk 1]
```diff
@@ -181,18 +199,45 @@ impl Action<'_> {
...
                     _ => Err("Invalid files: must be two elf or a pdb and a pe".into()),
                 }
             }
+            Self::ListArch => {
+                let buf = utils::read_file(&path_1);
+                mac::macho::MachoInfo::print_architectures(&buf, filename_1)?;
+
+                let buf = utils::read_file(&path_2);
+                mac::macho::MachoInfo::print_architectures(&buf, filename_2)
+            }
```
Comment (gabrielesvelto) [inline src/action.rs:240]: Lots of indentation here, better to move this to a separate function and possibly split the match arms in smaller functions.

File: src/common.rs [hunk 2]
```diff
@@ -14,17 +18,26 @@ pub(crate) enum FileType {
...
+            if let Ok(hint) = goblin::peek_bytes(start) {
+                match hint {
+                    Hint::Elf(_) => Self::Elf,
+                    Hint::Mach(_) | Hint::MachFat(_) => Self::Macho,
+                    Hint::PE => Self::Pe,
+                    _ => Self::Unknown,
+                }
+            } else {
+                Self::Unknown
+            }
```
Comment (gabrielesvelto) [inline src/common.rs:40]: Definitely move this to a separate function with a descriptive name.

File: src/mac/macho.rs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            let mut object = None;
+            for o in archive.objects() {
+                if let Ok(o) = o {
+                    if o.arch() == arch {
+                        object = Some(o);
+                        break;
+                    }
+                }
+            }
+            object
```
Comment (gabrielesvelto) [inline src/mac/macho.rs:43]: Couldn't this be done with `std::iter::Iterator::find`?

File: src/action.rs (hunk 1 above)
Comment (calixteman) [reply src/action.rs]: I'll fix that in another patch to better handle multiple file in entry.

**Pull request merged**